concordia/admin/__init__.py

```diff
@@ -233,6 +241,53 @@ def get_urls(self):
 
         return custom_urls + urls
 
+    def save_model(self, request, obj, form, change):
+
+        old_campaign = Project.objects.get(pk=obj.pk).campaign.slug
+        old_project = Project.objects.get(pk=obj.pk).slug
+
+        super().save_model(request, obj, form, change)
+        # call boto3 api to move project files from older campaign to newer campaign
+        # call boto3 api to update project name within same campaigns and transfer files
+        try:
+
+            new_campaign = obj.campaign.slug
+            new_project = obj.slug
+            # aws_profile = "AWSCrowdlocgov"
+            bucket_name = getattr(settings, "S3_BUCKET_NAME", None)
+            session = boto3.session.Session()
+            s3_resource = session.resource("s3")
+            top_level_bucket = s3_resource.Bucket(bucket_name)
+            filter_from = old_campaign + "/" + old_project
+            if old_campaign != new_campaign or old_project != new_project:
+
+                for obj in top_level_bucket.objects.filter(
+                    Prefix=filter_from + "/"
+                ).all():
+                    # print(obj.key.replace(old_campaign,new_campaign).replace(old_project,new_project))
+                    s3_resource.Object(
+                        bucket_name,
+                        obj.key.replace(old_campaign, new_campaign).replace(
+                            old_project, new_project
+                        ),
+                    ).copy_from(CopySource=bucket_name + "/" + obj.key)
+
+                res = []
+                bucket = s3_resource.Bucket(bucket_name)
+                for obj_version in bucket.object_versions.filter(
+                    Prefix=filter_from + "/"
+                ).all():
+                    res.append(
+                        {"Key": obj_version.object_key, "VersionId": obj_version.id}
+                    )
+                    # print(res)
+
+                if res != []:
+                    bucket.delete_objects(Delete={"Objects": res})
+
+        except Exception as error_message:
+            print(error_message)
```

Comment:
Exceptions should be properly handled and not printed - at a minimum log it and then raise it up to the calling method

Reply:
put it as an error in the logger and raised the exception.